GramAddict/core/views.py

```diff
@@ -702,5 +750,49 @@ def _navigateToTab(self, tab: ProfileTabs):
             button.click()
 
 
+class CurrentStoryView:
+    def __init__(self, device: DeviceFacade):
+        self.device = device
+
+    def getStoryFrame(self):
+        return self.device.find(
+            resourceId="com.instagram.android:id/reel_viewer_image_view",
+            className="android.widget.FrameLayout",
+        )
+
+    def getUsername(self):
+        reel_viewer_title = self.device.find(
+            resourceId="com.instagram.android:id/reel_viewer_title",
+            className="android.widget.TextView",
+        )
+        return "" if not reel_viewer_title.exists() else reel_viewer_title.get_text()
+
+    def getTimestamp(self):
+        reel_viewer_timestamp = self.device.find(
+            resourceId="com.instagram.android:id/reel_viewer_timestamp",
+            className="android.widget.TextView",
+        )
+        if reel_viewer_timestamp.exists():
+            timestamp = reel_viewer_timestamp.get_text().strip()
+            value = int(re.sub("[^0-9]", "", timestamp))
+            if timestamp[-1] == "s":
+                return datetime.timestamp(
+                    datetime.datetime.now() - datetime.timedelta(seconds=value)
+                )
+            elif timestamp[-1] == "m":
+                return datetime.timestamp(
+                    datetime.datetime.now() - datetime.timedelta(minutes=value)
+                )
+            elif timestamp[-1] == "h":
+                return datetime.timestamp(
+                    datetime.datetime.now() - datetime.timedelta(hours=value)
+                )
+            else:
+                return datetime.timestamp(
+                    datetime.datetime.now() - datetime.timedelta(days=value)
+                )
+        return ""
```

Comment:
Inconsistent return of string. I think would be better to return None. Makes more sense to expect a Date or None, not Date and String.

Reply:
I couldn't find where this was actually used - I don't think it is - but yeah good call.